src/main/scala/edu/berkeley/ce/rockslicing/RockSlicer.scala

```diff
@@ -33,21 +32,38 @@ object RockSlicer {
       // Error message will already be printed out by InputProcessor
       System.exit(-1)
     }
-    val (globalOrigin, rockVolume, joints) = inputOpt.get
-    val starterBlocks = Seq(Block(globalOrigin, rockVolume))
+    val (globalOrigin, boundingBox, rockVolumeInputs, jointSetInputs) = inputOpt.get
+    val generatedInput = JointGenerator(globalOrigin, boundingBox, rockVolumeInputs, jointSetInputs)
+    val starterBlocks = Seq(Block(globalOrigin, generatedInput.rockVolume))
 
     // Generate a list of initial blocks before RDD-ifying it
-    val seedBlocks = if (arguments.numProcessors > 1) {
-      val processorJoints = LoadBalancer.generateProcessorJoints(starterBlocks.head, arguments.numProcessors)
-      processorJoints.foldLeft(starterBlocks) { (currentBlocks, joint) =>
-        currentBlocks.flatMap(_.cut(joint))
+    val (seedJoints, remainingJoints) = if (arguments.numProcessors > 1) {
+      // Check if at least one of the input joint sets is persistent
+      val persistentCheck = jointSetInputs exists { jointSet =>
+        jointSet.persistence == 100
+      }
+
+      if (!persistentCheck) {
```

Comment:
Let's avoid declaring a `val` separately like this and change the code to something more like
```scala
if (jointSetInputs.forAll(_.persistence != 100))
```